cortex-m/src/asm.rs

```diff
@@ -244,6 +244,35 @@ pub unsafe fn semihosting_syscall(mut nr: u32, arg: u32) -> u32 {
     nr
 }
 
+/// Switch to unprivileged mode.
+///
+/// Sets CONTROL.SPSEL (setting the program stack to be the active
+/// stack) and CONTROL.nPRIV (setting unprivileged mode), updates the
+/// program stack pointer to the address in `psp`, then jumps to the
+/// address in `entry`.
+///
+/// # Safety
+///
+/// `psp` and `entry` must point to valid stack memory and executable
+/// code, respectively. `psp` must be 8 bytes aligned and point to
+/// stack top as stack grows towards lower addresses.
```

Comment:
Should it also be a safety requirement that the stack should be large enough so it doesn't overflow?

(It would be even better if we could set PSPLIM here to avoid this safety issue, but that's only available on a subset of the cores supported by the cortex-m crate.)

Reply:
Yes the comments should mention that. Setting PSPLIM would be a good follow-up PR.